addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -2438,6 +2444,50 @@ var SnippetsMenu = Widget.extend({
             }],
         }).open();
     },
+    /**
+     * @private
+     */
+    _onRenameBtnClick: function (ev) {
+        const $snippet = $(ev.target).closest('.oe_snippet');
+        const snippetName = $snippet.attr('name');
+        const confirmText = _t('Confirm');
+        const cancelText = _t('Cancel');
+        const $input = $(`
+            <we-input class="o_we_user_value_widget w-100 mx-1">
+                <div>
+                    <input type="text" autocomplete="chrome-off" value="${snippetName}" class="text-left"/>
+                    <we-button class="o_confirm_btn fa fa-check" title="${confirmText}"/>
+                    <we-button class="o_cancel_btn fa fa-times" title="${cancelText}"/>
+                </div>
+            </we-input>
+        `);
+        $snippet.find('we-button').remove();
+        $snippet.find('span.oe_snippet_thumbnail_title').replaceWith($input);
+        setTimeout(() => {
+            const $textInput = $input.find('input');
+            $textInput.focus();
+            $textInput.select();
+            $snippet.find('.oe_snippet_thumbnail').addClass('o_we_already_dragging'); // prevent drag
+            $input.find('.o_confirm_btn').click(async () => {
+                const name = $textInput.val();
+                if (name !== snippetName) {
+                    await this._rpc({
+                        model: 'ir.ui.view',
+                        method: 'rename_snippet',
+                        kwargs: {
+                            'name': name,
+                            'view_id': parseInt(ev.target.dataset.snippetId),
+                            'template_key': this.options.snippets,
+                        },
+                    });
+                }
+                await this._loadSnippetsTemplates(name !== snippetName);
+            });
+            $input.find('.o_cancel_btn').click(async () => {
+                await this._loadSnippetsTemplates(false);
+            });
```

Comment:
Should we account for the async parts in there by adding them to our edition Mutex queue? Not really sure it is needed but better safe than sorry I think.

Reply:
Added the mutex + loading effect on the RPC call.
It cannot be put around both the RPC and the `loadSnippetsTemplate` because that second part also uses it and the current implementation of the mutex is not re-entrant.